apps/payments/app/[locale]/(hosted)/paymentSetup/edit/[request_id]/page.tsx

```diff
@@ -0,0 +1,269 @@
+import { getTranslations } from "next-intl/server";
+import { getPaymentRequestDetails } from "../../db";
+import { PgSessions } from "auth/sessions";
+import { pgpool } from "../../../../../dbConnection";
+import { redirect } from "next/navigation";
+
+async function getRegisteredAccounts(userId: string, providerType: string) {
+  "use server";
+
+  const accountsQueryResult = await pgpool.query<
+    { provider_id: string; provider_name: string },
+    string[]
+  >(
+    `select provider_id, provider_name from payment_providers where user_id = $1 and provider_type = $2`,
+    [userId, providerType],
+  );
+
+  if (!accountsQueryResult.rowCount) {
+    return [];
+  }
+
+  return accountsQueryResult.rows;
+}
+
+async function editPayment(
+  userId: string,
+  paymentRequestId: string,
+  formData: FormData,
+) {
+  "use server";
+
+  const amountAsString = formData.get("amount")?.toString() ?? "";
+  // JS sucks at handling money
+  const amount = Math.round(parseFloat(amountAsString) * 100);
+
+  const openBankingAccount = formData.get("openbanking-account")?.toString();
+  const bankTransferAccount = formData.get("banktransfer-account")?.toString();
+  const stripeAccount = formData.get("stripe-account")?.toString();
+
+  if (!openBankingAccount && !bankTransferAccount && !stripeAccount) {
+    throw new Error("Failed to create payment");
+  }
+  const data = {
+    title: formData.get("title")?.toString(),
+    description: formData.get("description")?.toString(),
+    reference: formData.get("reference")?.toString(),
+    amount,
+    redirectUrl: formData.get("redirect-url")?.toString(),
+    allowAmountOverride: formData.get("allowAmountOverride") === "on",
+  };
+
+  const client = await pgpool.connect();
+
+  try {
+    await client.query("BEGIN");
+    const result = await client.query(
+      `update payment_requests 
+    set title = $1, description = $2, reference = $3, amount = $4, redirect_url = $5, allow_amount_override = $6 
+    where payment_request_id = $7 and user_id = $8`,
+      [
+        data.title,
+        data.description,
+        data.reference,
+        data.amount,
+        data.redirectUrl,
+        data.allowAmountOverride,
+        paymentRequestId,
+        userId,
+      ],
+    );
+    // Here it stops for whatever reason...
+    console.log(result);
+
+    redirect(`/paymentSetup/requests/${paymentRequestId}`);
+  } catch (e) {
+    await client.query("ROLLBACK");
+    console.log(e);
+  }
+
+  return;
```

Comment:
Looks like `redirect()` throws an error that Next then handles to stop code execution, so we need to move it outside of the try catch block. I think we can also remove `user_id` from the query, we just need `payment_request_id`

```suggestion
    await client.query(
      `update payment_requests 
    set title = $1, description = $2, reference = $3, amount = $4, redirect_url = $5, allow_amount_override = $6 
    where payment_request_id = $7`,
      [
        data.title,
        data.description,
        data.reference,
        data.amount,
        data.redirectUrl,
        data.allowAmountOverride,
        paymentRequestId,
      ],
    );
    await client.query("COMMIT");
  } catch (e) {
    await client.query("ROLLBACK");
    throw e;
  }
  redirect(`/paymentSetup/requests/${paymentRequestId}`);
```

Reply:
Good catch! At this step we can remove the user_id but when we create the APIs we need to keep that in mind. An user must be able to update only its requests